plugins/dependencies/src/main/kotlin/dependencies/Http4k.kt

```diff
@@ -4,103 +4,106 @@ private val GROUP = "org.http4k"
 
 object Http4k : DependencyGroup(
     group = GROUP,
-    usePlatformConstraints = true,
+    usePlatformConstraints = false,
     rawRule = """
         org.http4k:http4k-*
             ^^^^^^
     """.trimIndent()
 ) {
-    val bom = "org.http4k:http4k-bom:_"
-
-    val aws = module("http4k-aws")
-    val cloudnative = module("http4k-cloudnative")
-    val contract = module("http4k-contract")
-    val core = module("http4k-core")
-    val graphql = module("http4k-graphql")
-    val incubator = module("http4k-incubator")
-    val jsonrpc = module("http4k-jsonrpc")
-    val metricsMicrometer = module("http4k-metrics-micrometer")
-    val multipart = module("http4k-multipart")
-    val opentelemetry = module("http4k-opentelemetry")
-    val realtimeCore = module("http4k-realtime-core")
-    val resilience4j = module("http4k-resilience4j")
-    val securityOauth = module("http4k-security-oauth")
+    val bom: String get() {
+        usePlatformConstraints = true
+        return "org.http4k:http4k-bom:_"
+    }
+
+    val aws get() = module("http4k-aws")
+    val cloudnative get() = module("http4k-cloudnative")
+    val contract get() = module("http4k-contract")
+    val core get() = module("http4k-core")
+    val graphql get() = module("http4k-graphql")
+    val incubator get() = module("http4k-incubator")
+    val jsonrpc get() = module("http4k-jsonrpc")
+    val metricsMicrometer get() = module("http4k-metrics-micrometer")
+    val multipart get() = module("http4k-multipart")
+    val opentelemetry get() = module("http4k-opentelemetry")
+    val realtimeCore get() = module("http4k-realtime-core")
+    val resilience4j get() = module("http4k-resilience4j")
+    val securityOauth get() = module("http4k-security-oauth")
 
     val client = Client
 
     object Client : DependencyGroup(GROUP, usePlatformConstraints = true) {
-        val apache = module("http4k-client-apache")
-        val apacheAsync = module("http4k-client-apache-async")
-        val apache4 = module("http4k-client-apache4")
-        val apache4Async = module("http4k-client-apache4-async")
-        val jetty = module("http4k-client-jetty")
-        val okhttp = module("http4k-client-okhttp")
-        val websocket = module("http4k-client-websocket")
+        val apache get() = module("http4k-client-apache")
+        val apacheAsync get() = module("http4k-client-apache-async")
+        val apache4 get() = module("http4k-client-apache4")
+        val apache4Async get() = module("http4k-client-apache4-async")
+        val jetty get() = module("http4k-client-jetty")
+        val okhttp get() = module("http4k-client-okhttp")
+        val websocket get() = module("http4k-client-websocket")
```

Comment:
We could also use property delegation, with inline getters and setters to allow the `KProperty` to be optimized out once we compile against Kotlin 1.4 (and drop support for Gradle 6).